### PR TITLE
implement copy by value functionality

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
It is now possible to copy component descriptors and referenced component artifacts using the component-cli.
```

### CODE REVIEW

**Comment by schrodit** [comment]
/invite @jschicktanz @enrico-kaack-comp 

@enrico-kaack-comp can you check if the gardener bom can be transported

**Comment by enrico-kaack-comp** [comment]
Suggestion: Show some kind of Progress (not sure how much you show right now, I canceled and used -v 10)

File: pkg/commands/componentarchive/remote/copy.go [hunk 1]
```diff
@@ -159,18 +177,29 @@ type Copier struct {
 	Recursive bool
 	// Force forces an overwrite in the target registry if the component descriptor is already uploaded.
 	Force bool
+	// CopyByValue defines if all oci images and artifacts should be copied by value or reference.
+	// LocalBlobs are still copied by value.
+	CopyByValue bool
+	// SourceArtifactRepository is the source repository for oci artifacts.
+	// This value is only relevant if the artifacts are copied by value.
```
[the remark this answers is not in the document]
Comment (schrodit) [reply pkg/commands/componentarchive/remote/copy.go:184]: > Shouldn't it be exposed via the CLI flags?
how else should we get that value?

[the remark this answers is not in the document]
Comment (schrodit) [reply pkg/commands/componentarchive/remote/copy.go:184]: > Shouldn't it be exposed via the CLI flags?

yes good point added it as flag

[the remark this answers is not in the document]
Comment (jschicktanz) [reply pkg/commands/componentarchive/remote/copy.go:184]: Yeah :) but the CLI flag is still missing? or did I oversee it?

[the remark this answers is not in the document]
Comment (schrodit) [reply pkg/commands/componentarchive/remote/copy.go:194]: you replied faster than I was able to update my comment😅
I exposed the source repository as flag

File: pkg/commands/componentarchive/remote/copy.go [hunk 2]
```diff
@@ -190,40 +229,107 @@ func (c *Copier) Copy(ctx context.Context, name, version string) error {
...
+				log.V(7).Info("skip oci artifact copy by value", "resource", res.Name)
+				continue
+			}
+			ociRegistryAcc := &cdv2.OCIRegistryAccess{}
+			if err := res.Access.DecodeInto(ociRegistryAcc); err != nil {
+				return fmt.Errorf("unable to decode resource %s: %w", res.Name, err)
+			}
+
+			keepOrigHost := true
+			if c.ConvertToRelativeOCIReferences {
```
[the remark this answers is not in the document]
Comment (schrodit) [reply pkg/commands/componentarchive/remote/copy.go:271]: to my understanding we do not want that.
But now I'm also not sure if we want to keep the original registry every time.
Internally we only keep the original registry if we copy from open source registries but if we would do that every time we would have very long images references as every copy step would add its own registry

**Pull request merged**